firstvoices/backend/serializers/language_serializers.py

```diff
@@ -0,0 +1,68 @@
+from django.db.models import Prefetch
+from django.db.models.functions import Upper
+from rest_framework import serializers
+
+from backend.models import Language
+from backend.models.sites import Site, SiteFeature
+from backend.serializers.site_serializers import SiteSummarySerializer
+from backend.views.utils import get_select_related_media_fields
+
+
+class LanguageSerializer(serializers.Serializer):
+    """
+    Serializes basic details about a Language, including a list of visible (Public or Members) Sites for that Language.
+    """
+
+    language = serializers.CharField(source="title", read_only=True)
+    language_code = serializers.CharField(read_only=True)
+    sites = SiteSummarySerializer(many=True, read_only=True)
+
+    class Meta:
+        model = Language
+        fields = ("language", "language_code", "sites")
+
+    @classmethod
+    def make_queryset_eager(cls, queryset, visible_sites):
+        """Add prefetching as required by this serializer"""
+        return queryset.order_by(Upper("title")).prefetch_related(
+            Prefetch(
+                "sites",
+                queryset=visible_sites.order_by(Upper("title")).select_related(
+                    *get_select_related_media_fields("logo")
+                ),
+            ),
+            Prefetch(
+                "sites__sitefeature_set",
+                queryset=SiteFeature.objects.filter(is_enabled=True),
+            ),
+        )
+
+
+class MoreSiteSerializer(serializers.Serializer):
+    """
```

Comment:
New class containing formatting and prefetching that used to be in the LanguageViewSet